packages/compass-web/scripts/sync-dist-to-mms.mts

```diff
@@ -0,0 +1,183 @@
+let distWatcher: fs.FSWatcher | undefined;
+let webpackWatchProcess: child_process.ChildProcess | undefined;
+
+const tmpDir = path.join(
+  os.tmpdir(),
+  `mongodb-js--compass-web-${Date.now().toString(36)}`
+);
+const srcDir = path.resolve(import.meta.dirname, '..', 'dist');
+const destDir = path.dirname(
+  child_process.execFileSync(
+    'node',
+    ['-e', "console.log(require.resolve('@mongodb-js/compass-web'))"],
+    { cwd: process.env.MMS_HOME, encoding: 'utf-8' }
+  )
+);
+
+fs.mkdirSync(srcDir, { recursive: true });
+// Create a copy of current dist that will be overridden by link, we'll restore
+// it when we are done
+fs.mkdirSync(tmpDir, { recursive: true });
+fs.cpSync(destDir, tmpDir, { recursive: true });
+
+const failProofRunner = () =>
+  new (class FailProofRunner extends Array {
+    append(...fns: any[]) {
+      this.push(...fns);
+      return this;
+    }
+
+    run() {
+      const errors = this.map((f) => {
+        try {
+          f();
+        } catch (e) {
+          return e;
+        }
+      }).filter((e) => e);
+
+      if (errors.length) {
+        fs.writeSync(
+          process.stdout.fd,
+          util.inspect(errors, { depth: 20 }) + '\n'
+        );
+      }
+
+      return errors.length;
+    }
+  })();
+
+function cleanup(signalName: NodeJS.Signals): void {
+  console.log(`\nReceived ${signalName}, cleaning up...`);
+  const errorCount = failProofRunner()
+    .append(() => distWatcher?.close())
+    .append(() => webpackWatchProcess?.kill(signalName))
+    .append(() => devServer?.kill(signalName))
+    .append(() => fs.cpSync(tmpDir, destDir, { recursive: true }))
+    .append(() => fs.rmSync(tmpDir, { recursive: true, force: true }))
+    .run();
+  fs.writeSync(process.stdout.fd, 'Exit compass-web sync...\n');
+  process.exit(errorCount);
+}
+
+// Set up signal handlers immediately
+process.on('SIGINT', () => cleanup('SIGINT'));
+process.on('SIGTERM', () => cleanup('SIGTERM'));
+
+async function isDevServerRunning(
+  port: number,
+  host: string = '127.0.0.1'
+): Promise<boolean> {
+  try {
+    return (
+      await fetch(`http://${host}:${port}`, {
+        method: 'HEAD',
+        signal: AbortSignal.timeout(3000),
+      })
+    ).ok;
+  } catch (error) {
+    return false;
+  }
+}
+
+if (!(await isDevServerRunning(8081))) {
+  console.log('mms dev server is not running... launching!');
+
+  const { engines } = JSON.parse(
+    await asyncFs.readFile(
+      path.join(process.env.MMS_HOME, 'package.json'),
+      'utf8'
+    )
+  );
+  const pnpmVersion = engines.pnpm ?? 'latest';
+
+  const halfRamMb = Math.min(
+    Math.floor(os.totalmem() / 2 / 1024 / 1024),
+    16384
+  );
+  // Merge with existing NODE_OPTIONS if present
+  const existingNodeOptions = process.env.NODE_OPTIONS ?? '';
+  const mergedNodeOptions = [
+    `--max_old_space_size=${halfRamMb}`,
+    existingNodeOptions,
+  ]
+    .filter(Boolean)
+    .join(' ');
+
+  devServer = child_process.spawn(
```

Comment:
If we're not adding some sort of explicit way to disable strict engine check here, would be nice to monitor for the error and print something along the lines that the check can be disabled with `NPM_CONFIG_ENGINE_STRICT=false` flag